src/DevBetterWeb.Core/Specs/DailyCheckByDateSpec.cs

```diff
@@ -0,0 +1,15 @@
+﻿using Ardalis.Specification;
+using DevBetterWeb.Core.Entities;
+using System;
+
+namespace DevBetterWeb.Core.Specs
+{
+  public class DailyCheckByDateSpec : Specification<DailyCheck>, ISingleResultSpecification
+  {
+    public DailyCheckByDateSpec(DateTime date)
+    {
+      Query.Where(dc => dc.Date.Date.Equals(date.Date))
```

Comment:
Just checking, do you need `.dc.Date.Date` here?

Reply:
yes. the first .date is the property, the second .date tells it to ignore whatever the time is on the datetime

Reply:
Give the property a more descriptive name. I think BaseDomainEvent already has a date on it, but since this is the one you want to record with the "DailyCheck" perhaps it should be "CheckInitiatedDateTime"?

It would probably be good to store the full date time value with the DailyCheck, which I think you're doing. In that case, it's usually more clear to use `DateTime` in the property name rather than `Date` just to make that obvious.